base/abstractarray.jl

```diff
@@ -940,6 +940,11 @@ function copyto!(B::AbstractVecOrMat{R}, ir_dest::AbstractRange{Int}, jr_dest::A
     return B
 end
 
+function copyto_axcheck!(dest, src)
```

Comment:
This likely needs to be `@inline`d to make use of the `@boundscheck` block, but I'm not sure I'd worry much about that in the first place since you can always just call `copyto!` instead of `@inbounds copyto_axcheck!`.